scripts/linters/pylint_extensions_test.py

```diff
@@ -2370,6 +2370,870 @@ def test_checks_single_char_and_newline_eof(self):
             temp_file.close()
 
 
+class TypeIgnoreCommentCheckerTests(unittest.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.checker_test_object = testutils.CheckerTestCase()
+        self.checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.TypeIgnoreCommentChecker)
+        self.checker_test_object.setup_method()
+
+    def test_type_ignore_used_without_comment_raises_error(self):
+        node_function_with_type_ignore_only = astroid.scoped_nodes.Module(
+            name='test',
+            doc='Custom test'
+        )
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+
+        with utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                suggestion.change.new_value = (  # type: ignore[attr-defined]
+                    new_content
+                ) #@
+                """
+            )
+        node_function_with_type_ignore_only.file = filename
+
+        self.checker_test_object.checker.visit_module(
+            node_function_with_type_ignore_only
+        )
+        message = testutils.Message(
+            msg_id='mypy-ignore-used',
+            line=2,
+            node=node_function_with_type_ignore_only
+        )
+        with self.checker_test_object.assertAddsMessages(message):
+            temp_file.close()
```

Comment:
I'm doing this previously because in documentation we have an example where we are asserting on a closed temporary file.
https://github.com/oppia/oppia/wiki/Custom-Pylint-checks#tests-for-hello-world-token-checker

But now, we are asserting on `visit_module()` -- but i didn't removed `temp_file.close()` because `temp_file.close()` allow us to close and delete the temporary file that was created using `tempfile.NamedTemporaryFile()`. If I'm not wrong.